rkt/image/fetchers/common.go

```diff
@@ -0,0 +1,52 @@
+package fetchers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type Headers map[string][]string
+
+type InsecureOpts struct {
+	AllowHTTP      bool `json:"allow_http"`
+	SkipTLSCheck   bool `json:"skip_tls_check"`
+	SkipImageCheck bool `json:"skip_image_check"`
+}
+
+type Config struct {
+	Version       int                `json:"version"`
+	Scheme        string             `json:"scheme"`
+	Name          string             `json:"name"`
+	InsecureOpts  InsecureOpts       `json:"insecure"`
+	Debug         bool               `json:"debug"`
+	Headers       map[string]Headers `json:"headers"`
+	OutputACIPath string             `json:"output_aci_path"`
+	OutputASCPath string             `json:"output_asc_path"`
+}
+
```

Comment:
Beside we are talking about a scheme related to "distribution" or to a "transport", I'm always for one config type per scheme. Who know in future which transport will be used and which options will be required (bittorrent, ipfs?).